src/main/java/uk/ac/sanger/sccp/stan/service/ResultServiceImp.java

```diff
@@ -118,6 +132,35 @@ public void validateLabwareContents(Collection<String> problems, UCMap<Labware>
         }
     }
 
+    /**
+     * Checks the requested measurements on all the labware
+     * @param problems receptacle for problems
+     * @param labware map of labware from its barcode
+     * @param labwareResults the requests for each labware
+     * @return sanitised measurements for each labware barcode
+     */
+    public UCMap<List<SlotMeasurementRequest>> validateMeasurements(Collection<String> problems, UCMap<Labware> labware,
+                                                                    Collection<LabwareResult> labwareResults) {
+        if (labwareResults.stream().allMatch(lr -> lr.getSlotMeasurements().isEmpty())) {
+            return new UCMap<>(); // no measurements given
+        }
+        String coverageMeasurementName = MeasurementType.Tissue_coverage.friendlyName();
+        SlotMeasurementValidator val = slotMeasurementValidatorFactory.getSlotMeasurementValidator(List.of(coverageMeasurementName));
+        val.setValueSanitiser(coverageMeasurementName, coverageSanitiser);
+        UCMap<List<SlotMeasurementRequest>> sanMeasurements = new UCMap<>(labware.size());
+        for (LabwareResult lr : labwareResults) {
+            if (!lr.getSlotMeasurements().isEmpty()) {
+                Labware lw = labware.get(lr.getBarcode());
+                var sm = val.validateSlotMeasurements(lw, lr.getSlotMeasurements());
+                if (lw != null) {
```

Comment:
var sm = val.validateSlotMeasurements(lw, lr.getSlotMeasurements()); 
Just a thought- Is it better to put the above check as well in lw!==null condition?

Reply:
In this case we want validateSlotMeasurements to be run even if lw is null. It checks for additional problems, like the measurement name or value being invalid.